mllib/src/main/scala/org/apache/spark/ml/feature/FRegressionSelector.scala

```diff
@@ -0,0 +1,357 @@
+  extends Estimator[FRegressionSelectorModel] with FRegressionSelectorParams
+  with DefaultParamsWritable {
+
+  @Since("3.1.0")
+  def this() = this(Identifiable.randomUID("FRegressionSelector"))
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setNumTopFeatures(value: Int): this.type = set(numTopFeatures, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setPercentile(value: Double): this.type = set(percentile, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setFpr(value: Double): this.type = set(fpr, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setSelectorType(value: String): this.type = set(selectorType, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setFeaturesCol(value: String): this.type = set(featuresCol, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setOutputCol(value: String): this.type = set(outputCol, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setLabelCol(value: String): this.type = set(labelCol, value)
+
+  @Since("3.1.0")
+  override def fit(dataset: Dataset[_]): FRegressionSelectorModel = {
+    transformSchema(dataset.schema, logging = true)
+    val input: RDD[LabeledPoint] =
+      dataset.select(col($(labelCol)).cast(DoubleType), col($(featuresCol))).rdd.map {
+        case Row(label: Double, features: Vector) =>
+          LabeledPoint(label, features)
+      }
+    val FTestResult = FRegressionTest.test_regression(dataset, getFeaturesCol, getLabelCol)
+      .zipWithIndex
+    val features = $(selectorType) match {
+      case "numTopFeatures" =>
+        FTestResult
+          .sortBy { case (res, _) => res.pValue }
+          .take(getNumTopFeatures)
+      case "percentile" =>
+        FTestResult
+          .sortBy { case (res, _) => res.pValue }
+          .take((FTestResult.length * getPercentile).toInt)
+      case "fpr" =>
+        FTestResult
+          .filter { case (res, _) => res.pValue < getFpr }
+      case errorType =>
+        throw new IllegalStateException(s"Unknown FRegressionSelector Type: $errorType")
+    }
+    val indices = features.map { case (_, index) => index }
+    copyValues(new FRegressionSelectorModel(uid, indices).setParent(this))
+  }
+
+  @Since("3.1.0")
+  override def transformSchema(schema: StructType): StructType = {
+    SchemaUtils.checkColumnType(schema, $(featuresCol), new VectorUDT)
+    SchemaUtils.checkNumericType(schema, $(labelCol))
+    SchemaUtils.appendColumn(schema, $(outputCol), new VectorUDT)
+  }
+
+  @Since("3.1.0")
+  override def copy(extra: ParamMap): FRegressionSelector = defaultCopy(extra)
+}
+
+@Since("3.1.0")
+object FRegressionSelector extends DefaultParamsReadable[FRegressionSelector] {
+
+  @Since("3.1.0")
+  override def load(path: String): FRegressionSelector = super.load(path)
+}
+
+/**
+ * Model fitted by [[FRegressionSelector]].
+ */
+@Since("3.1.0")
+final class FRegressionSelectorModel private[ml] (
+    @Since("3.1.0") override val uid: String,
```

Comment:
I found that `f_regression` in scikit-learn will return both arrays of F-values and P-values, can we also add them to  `FRegressionSelectorModel`?